cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -109,13 +111,46 @@ class column_wrapper {
  *`[begin,end)`
  **/
 template <typename Element, typename InputIterator>
-rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
+rmm::device_buffer make_elements(
+  InputIterator begin,
+  InputIterator end,
+  typename std::enable_if<
+    std::is_convertible<typename std::iterator_traits<InputIterator>::value_type,
+                        Element>::value>::type* = nullptr)
 {
   static_assert(cudf::is_fixed_width<Element>(), "Unexpected non-fixed width type.");
   thrust::host_vector<Element> elements(begin, end);
   return rmm::device_buffer{elements.data(), elements.size() * sizeof(Element)};
 }
 
+/**
+ * @brief Creates a `device_buffer` containing the elements in the range
+ * `[begin,end)`.
+ *
+ * This is applicable for types that do not allow implicit construction of `Element` from the
```

Comment:
We should not be allowing non-implicit conversion like this. It is the callers responsibility to do the explicit conversion so that they are aware that it is taking place. That's the whole point of having explicit conversion is so that it is _explicit_.

Reply:
the initializer list constructor of the column_wrapper is used in a number of places where list of literal integer values are passed to construct `fixed_width_column_wrapper`. these are used in the context of typed test with arithmetic/integral types. the `make_elements` will try to assign the literal value to the target type - `timestamp`, `duration` etc. which will fail if the constructor is explicit. hence, instead of making this change in a number of places, this was done once here if the object cannot be implicitly constructed with the (integer) value.

Reply:
Yeah, I understand the use case. I'm saying that modifying `fixed_column_width_wrapper` is not the right place to make this change. Your use case is why something like this exists: https://github.com/rapidsai/cudf/blob/b29dec4925154824261163d1c3f1a17e5789f153/cpp/tests/utilities/type_lists.hpp#L67

My point is that it should be a separate, explicit function that requires the user to opt in to using it instead of handling it implicitly in `fixed_width_column_wrapper` where it can lead to surprising behavior.

Reply:
thank you. i did not know about that. i'll take a look.

Reply:
is it okay to specialize `make_elements` only for `chrono` types for explicit conversion from `int` types?

Reply:
@jrhemstad what do you think of the following factory method(s) to create the fixed width column wrappers? the intent of these method(s) are to create a fixed width column of type `TypeParam` by converting the input to `TypeParam`. this can be useful for scenarios that does not allow implicit conversion to `TypeParam`. this can also be useful later when we *only* allow construction of timestamps with durations.

the callers can explicitly opt in when they require such a conversion (which could be a large number of typed param tests that works on fixed width types). this may touch a number of files, and hence the request.

```
template <typename TypeParam, typename T>
auto make_fixed_width_column_with_type_param(std::initializer_list<T> const& init_list);

template <typename TypeParam, typename T>
auto make_fixed_width_column_with_type_param(std::initializer_list<T> const& init_list, 
                                             std::initializer_list<bool> const& validity);

...
```